clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1422,16 +1422,29 @@ protected void updateLatestInMemoryProcessedOffset(
         () -> OffsetRecord.NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY);
   }
 
-  protected void checkAndHandleUpstreamOffsetRewind(
+  protected static void checkAndHandleUpstreamOffsetRewind(
       PartitionConsumptionState partitionConsumptionState,
-      OffsetRecord offsetRecord,
       PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> consumerRecord,
       final long newUpstreamOffset,
-      final long previousUpstreamOffset) {
+      final long previousUpstreamOffset,
+      LeaderFollowerStoreIngestionTask ingestionTask) {
     if (newUpstreamOffset >= previousUpstreamOffset) {
       return; // Rewind did not happen
     }
-
+    if (!ingestionTask.isHybridMode()) {
```

Comment:
For a hybrid store, the misordered messages even before EOP can be problematic when later the log compaction is enabled, and that is the reason I only wanted to tolerate such issues in batch-only store.

Regarding hostname comparison instead of GUID comparison, based on the comments, originally, it was using GUID, and to support RP job, this function got changed to use hostname comparison.
I can image there might be multiple threads running in the same RP job instance, and messages with mixed GUID are expected in the same partition.
@shuhui-liu 
Can you help clarify ^?

Reply:
Well, let's take these two things separately...

For the initial condition, why is it a problem if we get the split brain issue pre-EOP? That part of the data should be idempotent... also, wouldn't these messages be skipped by DIV anyway? IOW, do we even get here at all pre-EOP?

For the hostname comparison, if RP needs it, then I guess we can leave it this way.

Reply:
I got your point, and I over-simplified the condition check.
Originally, I thought this check is good enough, but it doesn't cover the batch push for the hybrid stores.
There are 3 use cases before EOP:
1. Batch push to a batch-only store.
2. Batch push to a hybrid store.
3. RP job to a hybrid store.

For 1 & 2, split-brain issue before EOP is not a concern.
For 3, it might be a concern since RP job doesn't guarantee the uniqueness of keys produced in RP job.

I guess I can modify the condition check in the following way:
```
If the split brain issue happens before EOP and the input is sorted, we can ignore it.
```
So this condition check will cover 1 & 2.

Reply:
Pre-EOP && is sorted would be good, IMHO.

Reply:
We decided to remove this one from the scope of this PR and we will explore a longer term solution for the split-brain issue for the batch portion of the hybrid store.